### PR TITLE
Fix Rook Movement and Menu Bugs

### PR DESCRIPTION
**Pull request by CalebStadler**
Fixed the bug where if the rook tried to move on to its own square it would not throw and exception and fixed bug where the menu for captured pieces was not being updated properly at the start of a new game.

### CODE REVIEW

**Comment by hidayat0411** [comment]
Dengan bug pazel broo

File: PieceAbstract.java [hunk 1]
```diff
@@ -56,6 +56,12 @@ else if(pieceType == MainFrame.B_QUEEN)
         else
             return -1;
     }
```
Comment (nwestervelt) [inline PieceAbstract.java:58]: I suggest adding a comment here to match the other methods.
```Suggestion
    }
    //used to reset the captured piece counts back to 0
```

**Review by nwestervelt — CHANGES_REQUESTED** [review]
See my comment for the change I'm requesting.

Overall it looks good.

**Review by nwestervelt — APPROVED** [review]

**Review by hidayat0411 — COMMENTED** [review]
Helow gaes

**Pull request merged**